AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -128,7 +131,7 @@ class SyncMutationToCloudOperation: AsynchronousOperation {
                 request = GraphQLRequest<MutationSyncResult>.deleteMutation(of: model,
                                                                             modelSchema: modelSchema,
                                                                             where: graphQLFilter,
-                                                                            version: mutationEvent.version)
+                                                                            version: latestSyncMetadata?.version)
```

Comment:
should we check the optionality of `latestSyncMetadata` earlier than at the time of constructing the request? what should we do if it's nil?

Reply:
Makes sense, i forgot create mutations will be `nil` version. Seems like if query failed with error, we should still continue to send the mutation and let the conflict handler resolve the conflicts.